### ISSUE TITLE
SSL support is broken -> Fix provided

### ISSUE THREAD

**Issue by ralphwetzel** [fgallaire/wsgiserver#5]
Hi!
SSL support is broken in wsgiserver (and probably in CherryPy as well). I've tried to generate a fix & pull request, yet you've disabled the collaboration feature. Therefore find the changes attached:

line 1776:
```
        if hasattr(ssl, 'create_default_context'):
            self.context = ssl.create_default_context(
                purpose=ssl.Purpose.CLIENT_AUTH,
                cafile=self.certificate_chain
            )
            self.context.load_cert_chain(self.certificate, self.private_key)
```

line 1806:
```
            elif e.errno == ssl.SSL_ERROR_SSL:
                if e.args[1].find('http request') > 0:
                    # The client is speaking HTTP to an HTTPS server.
                    raise NoSSLError
                elif e.args[1].find('unknown protocol') > 0:
                    # The client is speaking some non-HTTP protocol.
                    # Drop the conn.
                    return None, {}
```

Hope this helps.

Ralph

**Comment by fgallaire** [comment]
Hello @ralphwetzel, can you precise your Python version ?

**Comment by ralphwetzel** [comment]
Hi!
Tested this with v3.6.0.

**Comment by ralphwetzel** [comment]
Hi again!
You're right; nothing was disabled. I just didn't get how to create a pull request.
Yet it is done [now](https://github.com/fgallaire/wsgiserver/pull/6)...
Regards, Ralph

**Comment by horribleperson** [comment]
Is this fix verified? Will it be merged to master?

**Comment by fgallaire** [comment]
hello @horribleperson I have not yet evaluate this fix.

**Comment by webknjaz** [comment]
@ralphwetzel @horribleperson this all has been fixed in [cheroot (cheroot/ssl/builtin.py)](https://github.com/cherrypy/cheroot/blob/master/cheroot/ssl/builtin.py), which is a part of CherryPy moved into separate package a while ago.

**Comment by fgallaire** [comment]
hello @webknjaz do you want to contribute and help to fix wsgiserver ?

**Comment by webknjaz** [comment]
No, I'm maintaining upstream cherrypy and cheroot. Just saw some issue referring ours and followed the link to inform that related one is closed.

**Comment by webknjaz** [comment]
It's been changed lots of times, just compare the code.

**Comment by plainas** [comment]
What exactly was broken? Were these changes applied after all?

**Comment by webknjaz** [comment]
@plainas SSL error processing

**Comment by fgallaire** [comment]
Fixed by #6
